code/go/0chain.net/smartcontract/storagesc/block_reward.go

```diff
@@ -2,10 +2,172 @@ package storagesc
+			"cannot get smart contract configurations: "+err.Error())
+	}
+
+	if conf.BlockReward.BlockReward == 0 {
+		return nil
+	}
+
+	bbr, err := getBlockReward(conf.BlockReward.BlockReward, balances.GetBlock().Round,
+		conf.BlockReward.BlockRewardChangePeriod, conf.BlockReward.BlockRewardChangeRatio,
+		conf.BlockReward.BlobberWeight)
+	if err != nil {
+		return common.NewError("blobber_block_rewards_failed",
+			"cannot get block rewards: "+err.Error())
+	}
+
+	allBlobbers, err := getActivePassedBlobbersList(balances, conf.BlockReward.TriggerPeriod)
+	if err != nil {
+		return common.NewError("blobber_block_rewards_failed",
+			"cannot get all blobbers list: "+err.Error())
+	}
+
+	hashString := encryption.Hash(balances.GetTransaction().Hash + balances.GetBlock().PrevHash)
+	var randomSeed int64
+	randomSeed, err = strconv.ParseInt(hashString[0:15], 16, 64)
+	if err != nil {
+		return common.NewError("blobber_block_rewards_failed",
+			"error in creating seed"+err.Error())
+	}
+	r := rand.New(rand.NewSource(randomSeed))
+
+	blobberPartition, err := allBlobbers.GetRandomSlice(r, balances)
+	if err != nil {
+		return common.NewError("blobber_block_rewards_failed",
+			"Error getting random partition: "+err.Error())
+	}
+
+	for _, b := range blobberPartition {
+		var sp *stakePool
+		var blobber partitions.BlobberRewardNode
+
+		err = blobber.Decode(b.Encode())
+		if err != nil {
+			return common.NewError("blobber_block_rewards_failed",
+				"can't decode blobber reward node: "+err.Error())
+		}
+		if sp, err = ssc.getStakePool(b.Name(), balances); err != nil {
+			return common.NewError("blobber_block_rewards_failed",
+				"can't get related stake pool: "+err.Error())
+		}
+
+		stake := float64(sp.stake())
+
+		gamma := maths.GetGamma(A, B, alpha, blobber.TotalData, blobber.DataRead)
+		zeta := maths.GetZeta(I, K, mu, float64(blobber.WritePrice), float64(blobber.ReadPrice))
+		qualifyingBlobberIds = append(qualifyingBlobberIds, blobber.ID)
+		stakePools = append(stakePools, sp)
+		stakeTotals = append(stakeTotals, stake)
+		totalQStake += stake
+		blobberWeight := (gamma*zeta*float64(blobber.SuccessChallenges) + 1) * stake
+		weight = append(weight, blobberWeight)
+		totalWeight += blobberWeight
+	}
+
+	if totalWeight == 0 {
+		totalWeight = 1
+	}
+
+	for i, qsp := range stakePools {
+		weightRatio := weight[i] / totalWeight
+		if weightRatio > 0 && weightRatio <= 1 {
+			reward := bbr * weightRatio
+			logging.Logger.Info("blobber_block_rewards_pass",
+				zap.Float64("reward", reward),
+				zap.String("blobber id", qualifyingBlobberIds[i]))
+
+			if err := qsp.DistributeRewards(reward, qualifyingBlobberIds[i], stakepool.Blobber, balances); err != nil {
+				return common.NewError("blobber_block_rewards_failed", "minting capacity reward"+err.Error())
+			}
+
+			qsp.Reward += state.Balance(reward) // do we need to do this?
+		} else {
+			logging.Logger.Error("blobber_bloc_rewards - error in weight ratio",
+				zap.Any("stake pool", qsp))
```

Comment:
We need to return error instead of continue for loop of reward calucation to avoid potential reward calculation issue.

Reply:
done